src/libslic3r/Slicing.cpp

```diff
@@ -314,6 +314,8 @@ std::vector<double> layer_height_profile_adaptive(const SlicingParameters& slici
         else if (layer_height_profile.back() > height && layer_height_profile.back() - height > LAYER_HEIGHT_CHANGE_STEP)
             height = layer_height_profile.back() - LAYER_HEIGHT_CHANGE_STEP;
 
+        height = std::round(height * 1000) / 1000.f;
```

Comment:
Actually rounding could be skipped.

Reply:
How so? Dont we still need to round the gcode output, not just the display output?

Reply:
I don't now. Let's to @SoftFever decide.

Reply:
Agreed, rounding here is not necessary.